### PR TITLE
Send Email with Code

### LINKED ISSUE #33: Send Email with Code

### PR DESCRIPTION
**Pull request by edwardsung4217**
#33 Send Email with Code
> Team member: Edward

Fixes #33 

## Expected Behavior
An endpoint that sends the user a code that allows them to reset their password has been added to the API

A new function called SaveResetCodeToUser adds the ability for the user to reset their password incase they forget it. The function will send the user an email with a code in which they will need as it is the verification.

### - Usage 
Make POST request to:
http://www.localhost:8000/login/email@example.com
Takes in user email, checks if user exist in the database, then sends them a password reset code that is sent to their email.

### - Evidence/Logs
I can't currently provide evidence since my sendGrid account got suspended since I accidentally exposed the API key when I committed it.

### - Considerations
For this issue it was just sending the email with code. The actual changing of password and saving the new password will be implemented in another issue.

### CODE REVIEW

**Comment by lazeroffmichael** [comment]
I am thinking that we should move towards some type of `/api/v1/` scheme, however this route should be fine for now, thank you.

**Review by lazeroffmichael — APPROVED** [review]

File: backend/models/usersModel.go
Comment (Rahgid) [inline backend/models/usersModel.go:1]: While we're updating this file, can we remove the "id" field? This is included by gorm.Model already with the same properties.

**Review by Rahgid — CHANGES_REQUESTED** [review]

File: backend/models/usersModel.go
Comment (Rahgid) [reply backend/models/usersModel.go:1]: The field is exactly the same, it's just redundant since it's already included by gorm.Model. 

image

Comment (edwardsung4217) [reply backend/models/usersModel.go:1]: Sounds good, ill delete it once im  available

Comment (edwardsung4217) [reply backend/models/usersModel.go:1]: It has been deleted, please approve now

**Review by Rahgid — APPROVED** [review]
Looks good for now, but please also update "EXAMPLEAPIKEY" with an environment variable once your account is reactivated.

**Pull request merged**